### PR TITLE
Add ability to shake fizzy drinks so they spray in peoples' faces

### PR DESCRIPTION
**Pull request by Tayrtahn**
## About the PR

Adds the ability to use the verb menu to Shake most closed solution containers. This makes a sound, pops up a message, and runs a short (1 second) do-after.
If the solution contains reagents that are fizzy, shaking will cause pressure to build up in the container.
Shaking the container repeatedly will further increase the pressure. It takes two shakes to fully agitate a soda can, and different reagents agitate at different rates.
If there's a mixture of reagents in a solution, the fizziness is weighted by each reagent's proportion.
Pressure will slowly decrease over time as the solution settles. This takes two minutes by default.
Opening the container has a chance to cause the contents to spray out, with the chance increasing with the pressure in the container.
When a solution sprays, it will splash onto anyone nearby then make a puddle on the ground.
Throwing the container will also increase the pressure when it lands, with a chance for the container to burst open and spray if the pressure is high enough.
You can shake non-fizzy solutions too, but it won't do anything other than make a sound and show a popup.
You can also shake the bartender's drink shaker. This doesn't do anything, but it's there for RP purposes. And it's called a shaker, after all.

TL;DR: Shake a soda can a few times, hand it to someone, and watch them get sprayed with soda.

## Why / Balance

Adds content for clowns, passengers, or others to pull low-deadliness pranks on other crew members. Could be more deadly with some creativity, so that's fun too. Also bartender RP.
Idea was originally suggested by Rampant Lotus in this Discord thread: https://discord.com/channels/310555209753690112/1205626775544266772

## Technical details

Adds ShakeableComponent, which adds the Shake verb. Shaking raises ShakeEvent on the entity, and it's up to other components to respond to that.

Completely overhauls PressurizedDrinkComponent, turning it into PressurizedSolutionComponent. It handles simulating the pressure, checking solutions for fizziness, and spraying the contents. TimeSpans are used for fizziness so no ticking is needed. Previously, PressurizedDrinkComponent just gave drinks flat a chance to burst open when thrown.

Almost everything is in Shared and supports prediction. Random rolls are done in advance to try to minimize mispredictions. Puddle creation is handled by the server and not predicted, because we don't have client entity spawn prediction.

I did move part of DrinkSystem to Shared. IsEmpty and DrinkVolume were moved since I needed them, and I moved OnExamined and HalfEmptyOrHalfFull as a bonus, so now the "Open" text in the examine description is predicted.

ReagentPrototype has a new float value: Fizziness. PressurizedDrinkSystem looks up the reagent prototypes and calculates the "fizzability" from each reagent's fizziness multiplied by the proportion of the solution it makes up. A fizzy solution is required for PressurizedDrinkSystem to track pressure.

## Media

Shaking some soda cans, spraying the player if held:

https://github.com/space-wizards/space-station-14/assets/85356/7044ecd5-2dc0-4d0e-8df3-826631b9e37e

Non-fizzy reagent solutions don't get fizzy:

https://github.com/space-wizards/space-station-14/assets/85356/fcfdcbf3-78bb-43a4-a5ec-db2a27b44c5f

There are sound effects for shaking and spraying, but these videos are silent. Blame Apple for not recording audio with their built-in screen capture tool.

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

PressurizedDrinkComponent is gone. Use PressurizedSolutionComponent on containers that should be able to spray when opened if the solution inside is agitated (even if they start with a non-fizzy solution, remember that they can be refilled and reclosed). Reagents now have a Fizziness property which defaults to 0.

**Changelog**

:cl:
- add: Sodas and other fizzy drinks can be shaken with the verbs menu to build up fizziness so they spray when opened.

### CODE REVIEW

**Comment by Tayrtahn** [comment]
> super awesome change, would it be possible to have pressure build up at a rate proportional to the % of fizzy liquids in a container? (so a solution with 1u soda and 99u water would probably not explode very easily)

Yeah, that definitely should be doable

**Comment by VektorZ1** [comment]
Is it possible to add a reaction requiring pressure or shaking? It would be really cool to have to actually shake some drinks to create them

**Comment by Kadeo64** [comment]
pressurizing reagents real

**Comment by Tayrtahn** [comment]
Currently planned changes/additions:
- Replace boolean Fizzy value of reagents with a 0-1 float Fizziness. Fizzability?
- Calculate overall solution fizzability proportional to reagent ratio.
- Scale the effect of shaking/agitating by the overall fizzability of the solution. This will also put a soft cap on fizziness of solutions with low fizzability, since the natural loss of fizziness over time will eventually be faster than can be gained by shaking.

> Is it possible to add a reaction requiring pressure or shaking? It would be really cool to have to actually shake some drinks to create them

I know nothing about how reactions are implemented, but I'll look into it. I imagine it shouldn't be too hard to set up a reaction listening for a ShakeEvent, but we'll see.

**Comment by Tayrtahn** [comment]
Note: if #25813 gets merged, this can be cleaned up quite a bit.

**Comment by Tayrtahn** [comment]
Marking as draft until I clean this up using the changes from #25813.

File: Content.Shared/Nutrition/Components/PressurizedSolutionComponent.cs [hunk 1]
```diff
@@ -0,0 +1,106 @@
+using Content.Shared.Nutrition.EntitySystems;
+using Robust.Shared.Audio;
+using Robust.Shared.GameStates;
+
+namespace Content.Shared.Nutrition.Components;
+
+/// <summary>
+/// Represents a solution container that can hold the pressure from a solution that
+/// gets fizzy when aggitated, and can spray the solution then opened or thrown.
```
Comment (Tayrtahn) [inline Content.Shared/Nutrition/Components/PressurizedSolutionComponent.cs:9]:
```suggestion
/// gets fizzy when aggitated, and can spray the solution when opened or thrown.
```

**Review by Tayrtahn — COMMENTED** [review]
Leaving myself some notes
